Shell/LineEditor.cpp

```diff
@@ -354,15 +388,43 @@ String LineEditor::get_line(const String& prompt)
                 }
 
                 String token = is_empty_token ? String() : String(&m_buffer[token_start], m_cursor - (size_t)token_start);
+                Vector<String> suggestions;
 
                 if (is_first_token)
-                    tab_complete_first_token(token);
+                    suggestions = tab_complete_first_token(token);
                 else
-                    tab_complete_other_token(token);
+                    suggestions = tab_complete_other_token(token);
+
+                if (m_times_tab_pressed > 1 && !suggestions.is_empty()) {
+                    size_t longest_name = 0;
+
+                    for (auto& suggestion : suggestions)
+                        longest_name = max(longest_name, suggestion.length());
+
+                    size_t num_printed = 0;
+                    putchar('\n');
+                    for (auto& suggestion : suggestions) {
+                        int new_pos = num_printed + suggestion.length() + (longest_name + 2 - suggestion.length());
+
+                        if (new_pos > m_num_columns) {
+                            putchar('\n');
+                            num_printed = 0;
+                        }
 
+                        num_printed += printf("%-*s", static_cast<int>(longest_name) + 2, suggestion.characters());
+                    }
+
+                    putchar('\n');
+                    write(STDOUT_FILENO, prompt.characters(), prompt.length());
+                    write(STDOUT_FILENO, m_buffer.data(), m_cursor);
+                }
+
+                suggestions.clear_with_capacity();
                 continue;
             }
 
+            m_times_tab_pressed = 0; // Safe to say if we get here, the user didn't press TAB
```

Comment:
One potential problem is if the user presses tab, then one of the keys that is detected before tab, and then tab again.
Try this:
`ca<tab><left><tab>`
It's a very minor issue but I thought I would mention it

Reply:
I get the same results on Linux? Or am I missing something??

Reply:
On Bash I get no output (because it counts 1 tab) but with this the 't' gets cut off when the new line is printed and I get suggestions for 'c'.

Reply:
Ahh I see why

`write(STDOUT_FILENO, m_buffer.data(), m_cursor);`